hazelcast/src/main/java/com/hazelcast/map/impl/eviction/EvictionOperator.java

```diff
@@ -113,6 +109,30 @@ public void removeEvictableRecords(RecordStore recordStore, int evictableSize, M
         }
     }
 
+    protected boolean tryEvict(Record record, RecordStore recordStore, boolean backup, long now) {
+        String mapName = recordStore.getName();
+        Data key = record.getKey();
+        Object value = record.getValue();
+
+        if (!evictIfNotLocked(key, recordStore, backup)) {
+            return false;
+        }
+
+        if (!backup) {
+            mapServiceContext.interceptAfterRemove(mapName, value);
+
+            boolean expired = recordStore.isExpired(record, now, false);
+            if (expired) {
+                // Fire EXPIRED event also because there is a possibility that
+                // evicted entry may be also an expired one.
```

Comment:
is this important? what if the entry is not expired now, but it's gonna be expired in 1s. then we are not firing the expire event anyway.